### PR TITLE
fix(github releases) include "latest" release by default

### PR DESCRIPTION
**Pull request by dduportal**
Fixup of #969

As per https://github.com/updatecli/updatecli/pull/969#issuecomment-1318989437, the "latest" release was excluded from the returned collection of releases which leds to regression on the manifests not updated to use "typeFilter".

This PR sets up "Latest" enabled by default (assuming it is a release).

@olblak What is the use case for "Latest" in the typeFilter exactly? I smell a LOT of incoming bugs depending on if "Latest" is a prerelease or not.

## Test

To test this pull request:

```yaml
---
name: Bump JDK11 version

scms:
  default:
    kind: github
    spec:
      user: "{{ .github.user }}"
      email: "{{ .github.email }}"
      owner: "{{ .github.owner }}"
      repository: "{{ .github.repository }}"
      token: "{{ requiredEnv .github.token }}"
      username: "{{ .github.username }}"
      branch: "{{ .github.branch }}"

sources:
  lastVersion:
    kind: githubrelease
    name: Get the latest Adoptium JDK11 version
    spec:
      owner: "adoptium"
      repository: "temurin11-binaries"
      token: "{{ requiredEnv .github.token }}"
      username: "{{ .github.username }}"
      versionfilter:
        kind: regex
        # jdk-11.0.12+7(https://github.com/adoptium/temurin11-binaries/releases/tag/jdk-11.0.12%2B7) is OK
        # jdk-11.0.16.1+1 (https://github.com/adoptium/temurin11-binaries/releases/tag/jdk-11.0.16.1%2B1) is OK
        pattern: "^jdk-11.(\\d*).(\\d*).(\\d*)(.(\\d*))+(\\d*)$"
```

should return `11.0.17+8` (e.g. the "latest" when writing these lines).

### Tradeoff

- Remove the "Latest" filter at all from the code (not clear what would be the use case)

### CODE REVIEW

**Comment by olblak** [comment]
Thanks for catching this regression. 

> Remove the "Latest" filter at all from the code (not clear what would be the use case)

GitHub release now allows a user to label a specific release as latest ([cfr](https://docs.github.com/en/repositories/releasing-projects-on-github/managing-releases-in-a-repository)). It's useful when you have multiple release line on a repository.

Let's say a project release the version 2.378 and then 2.361.4. The 2.361.4 was considered as the latest because released after 2.378.

This issue happened to me multiple time in the past

My initial intend was if latest set to true, we only pick the release flag "latest" otherwise we ignore that rule

**Comment by dduportal** [comment]
> My initial intend was if latest set to true, we only pick the release flag "latest" otherwise we ignore that rule

🤔 then, the implementation is not following this behavior as far as I can tell. Implementation filters results from request, but if you set "latest" to "true", then you might want to do a different request to GitHub (expecting it to answer only the "latest"). Hence my question about intent: I already know what is the "latest" request. But I'm asking for a use case with updatecli.


In order to help implementing / fixing, could you provide a set of examples of updatecli manifsts (source only) showing different result behaviors you have in mind please?

**Comment by olblak** [comment]
What do you think of this approach? 
I didn't tested it yet https://github.com/updatecli/updatecli/pull/989/files

**Comment by dduportal** [comment]
Oh, I see https://github.com/updatecli/updatecli/pull/989. Care to add tests please: it is really hard to follow up without docs/example/tests

**Comment by olblak** [comment]
I can't reproduce this issue.
Your manifest returns me 11.0.17+8 

I also tried using https://github.com/olblak/nocode/releases where I created a new stable release without the latest flag but without success

**Comment by olblak** [comment]
```
SOURCES
=======

lastVersion
-----------
Searching for version matching pattern "^jdk-11.(\\d*).(\\d*).(\\d*)(.(\\d*))+(\\d*)$"
✔ Github Release version "jdk-11.0.17+8" found matching pattern "^jdk-11.(\\d*).(\\d*).(\\d*)(.(\\d*))+(\\d*)$"
```

**Comment by olblak** [comment]
I would prefer fixing the code than enabling "latest" by default. 
Before introducing the releasefilter "latest", the default behavior was to retrieve the latest published release (without considering the flag "latest")

**Comment by dduportal** [comment]
```
➜  updatecli git:(73205ba) git rev-parse --verify HEAD 
73205ba66ac88e57963f4bde9e98013cee8ef5c9
➜  updatecli git:(73205ba) go build -o dist/updatecli                                                                                                           
➜  updatecli git:(73205ba) cat .tmp/updatecli.yaml 
---
name: Bump JDK11 version

scms:
  default:
    kind: github
    spec:
      user: "{{ .github.user }}"
      email: "{{ .github.email }}"
      owner: "{{ .github.owner }}"
      repository: "{{ .github.repository }}"
      token: "{{ requiredEnv .github.token }}"
      username: "{{ .github.username }}"
      branch: "{{ .github.branch }}"

sources:
  lastVersion:
    kind: githubrelease
    name: Get the latest Adoptium JDK11 version
    spec:
      owner: "adoptium"
      repository: "temurin11-binaries"
      token: "{{ requiredEnv .github.token }}"
      username: "{{ .github.username }}"
      versionfilter:
        kind: regex
        # jdk-11.0.12+7(https://github.com/adoptium/temurin11-binaries/releases/tag/jdk-11.0.12%2B7) is OK
        # jdk-11.0.16.1+1 (https://github.com/adoptium/temurin11-binaries/releases/tag/jdk-11.0.16.1%2B1) is OK
        pattern: "^jdk-11.(\\d*).(\\d*).(\\d*)(.(\\d*))+(\\d*)$"
➜  updatecli git:(73205ba) ./dist/updatecli --values .tmp/values.yaml --config .tmp/updatecli.yaml diff


+++++++++++
+ PREPARE +
+++++++++++

Loading Pipeline ".tmp/updatecli.yaml"

SCM repository retrieved: 1


++++++++++++++++++
+ AUTO DISCOVERY +
++++++++++++++++++



++++++++++++
+ PIPELINE +
++++++++++++



######################
# BUMP JDK11 VERSION #
######################


SOURCES
=======

lastVersion
-----------
Searching for version matching pattern "^jdk-11.(\\d*).(\\d*).(\\d*)(.(\\d*))+(\\d*)$"
✔ Github Release version "jdk-11.0.16.1+1" found matching pattern "^jdk-11.(\\d*).(\\d*).(\\d*)(.(\\d*))+(\\d*)$"


CHANGELOG:
----------

Release published on the 2022-08-19 17:25:17 +0000 UTC at the url https://github.com/adoptium/temurin11-binaries/releases/tag/jdk-11.0.16.1%2B1

Official Release of jdk-11.0.16.1+1

=============================

REPORTS:


- Bump JDK11 version:
        Source:
                ✔ [lastVersion] Get the latest Adoptium JDK11 version (kind: githubrelease)

Run Summary
===========
Pipeline(s) run:
  * Changed:    0
  * Failed:     0
  * Skipped:    1
  * Succeeded:  0
  * Total:      1
```

=> I don't mind closing this PR in favor of an issue

**Comment by dduportal** [comment]
> I would prefer fixing the code than enabling "latest" by default. Before introducing the releasefilter "latest", the default behavior was to retrieve the latest published release (without considering the flag "latest")

The problem with #969 is that it is buggy: by default, IF the "latest" release is marked as "Latest" in GitHub releases, then it is removed from the list returned by GitHub.

That is why my example fails: updatecli returns the "most recent element" from the filtered collection, but this filtered collection is stripped out of the "Latest GitHub release".
